sunpy/visualization/imageanimator.py

```diff
@@ -700,15 +730,49 @@ def plot_start_image(self, ax):
         extent = []
         # reverse because numpy is in y-x and extent is x-y
         for i in self.image_axes[::-1]:
-            extent += self.axis_ranges[i]
+            if self.non_regular_plot_axis is False and len(self.axis_ranges[i]) > 2:
+                self.non_regular_plot_axis = True
+            extent.append(self.axis_ranges[i][0])
+            extent.append(self.axis_ranges[i][-1])
 
         imshow_args = {'interpolation': 'nearest',
                        'origin': 'lower',
-                       'extent': extent,
-                       }
-
+                       'extent': extent}
         imshow_args.update(self.imshow_kwargs)
-        im = ax.imshow(self.data[self.frame_index], **imshow_args)
+
+        # If value along an axis is set with an array, generate a NonUniformImage
+        if self.non_regular_plot_axis:
+            # If user has inverted the axes, transpose the data so the dimensions match.
+            if self.image_axes[0] < self.image_axes[1]:
+                data = self.data[self.frame_index].transpose()
+            else:
+                data = self.data[self.frame_index]
+            # Initialize a NonUniformImage with the relevant data and axis values and
+            # add the image to the axes.
+            im = mpl.image.NonUniformImage(ax, **imshow_args)
+            im.set_data(self.axis_ranges[self.image_axes[0]],
+                        self.axis_ranges[self.image_axes[1]], data)
+            ax.add_image(im)
+            # Define the xlim and ylim bearing in mind that the pixel values along
+            # the axes are plotted in the middle of the pixel.  Therefore make sure
+            # there's half a pixel buffer either side of the ends of the axis ranges.
+            xlim = (self.axis_ranges[self.image_axes[0]][0] - \
+                    (self.axis_ranges[self.image_axes[0]][1] - \
+                     self.axis_ranges[self.image_axes[0]][0]) / 2.,
+                    self.axis_ranges[self.image_axes[0]][-1] + \
+                    (self.axis_ranges[self.image_axes[0]][-1] - \
+                     self.axis_ranges[self.image_axes[0]][-2]) / 2.)
+            ylim = (self.axis_ranges[self.image_axes[1]][0] - \
+                    (self.axis_ranges[self.image_axes[1]][1] - \
+                     self.axis_ranges[self.image_axes[1]][0]) / 2.,
+                    self.axis_ranges[self.image_axes[1]][-1] + \
+                    (self.axis_ranges[self.image_axes[1]][-1] - \
+                     self.axis_ranges[self.image_axes[1]][-2]) / 2.)
```

Comment:
This it not easy to read.  This would be easier to read and understand if you define something like xxx_axis = self.axis_ranges[self.image_axes[0]] and yyy_axis = self.axis_ranges[self.image_axes[1]] before these definitions.